packages/lib-classifier/src/components/Classifier/Classifier.spec.js

```diff
@@ -0,0 +1,507 @@
+          }
+          case '/project_roles': {
+            const project_roles = [{ roles }]
+            return Promise.resolve({ body: { project_roles }})
+          }
+          case '/subjects/queued': {
+            const subjects = [subjectSnapshot, ...Factory.buildList('subject', 9)]
+            return Promise.resolve({ body: { subjects }})
+          }
+        }
+      })
+      sinon.stub(panoptes, 'post').callsFake((...args) => {
+        return Promise.resolve({ headers: {}, body: { project_preferences: []}})
+      })
+      const checkBearerToken = sinon.stub().callsFake(() => Promise.resolve('mockAuth'))
+      const checkCurrent = sinon.stub().callsFake(() => Promise.resolve({ id: 123, login: 'mockUser' }))
+      const authClient = { ...defaultAuthClient, checkBearerToken, checkCurrent }
+      const client = { ...defaultClient, panoptes }
+      const store = RootStore.create({}, { authClient, client })
+      render(
+        <Classifier
+          classifierStore={store}
+          project={projectSnapshot}
+          workflowID={workflowSnapshot.id}
+          workflowSnapshot={workflowSnapshot}
+          workflowVersion={workflowSnapshot.version}
+        />,
+        {
+          wrapper: withStore(store)
+        }
+      )
+      store.subjectViewer.onSubjectReady()
+      taskTab = screen.queryByRole('tab', { name: 'TaskArea.task'})
+      tutorialTab = screen.queryByRole('tab', { name: 'TaskArea.tutorial'})
+      subjectImage = screen.queryByRole('img', { name: `Subject ${subjectSnapshot.id}` })
+      tabPanel = screen.queryByRole('tabpanel', { name: '1 Tab Contents'})
+      const task = workflowSnapshot.tasks.T0
+      const getAnswerInput = answer => within(tabPanel).queryByRole('radio', { name: answer.label })
+      taskAnswers = task.answers.map(getAnswerInput)
+    })
+
+    after(function () {
+      panoptes.get.restore()
+      panoptes.post.restore()
+    })
+
+    it('should have a task tab', function () {
+      expect(taskTab).to.be.ok()
+    })
+
+    it('should have a tutorial tab', function () {
+      expect(tutorialTab).to.be.ok()
+    })
+
+    it('should not have a subject image', function () {
+      expect(subjectImage).to.be.null()
+    })
+
+    describe('task answers', function () {
+      it('should not be displayed', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton).to.be.null()
+        })
+      })
+    })
+  })
+
+  describe('with showTutorial', function () {
+    let tutorialHeading
+
+    before(async function () {
+      const subject = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }] })
+      const steps = [
+        { content: "Hello" },
+        { content: "Step 2" }
+      ]
+      const workflowTutorial = TutorialFactory.build({ steps })
+      const store = mockStore({ subject })
+      store.tutorials.setResources([workflowTutorial])
+      store.tutorials.setActive(workflowTutorial.id)
```

Comment:
@eatyourgreens as I'm working through lib-classifier's specs, and converting them to run with Vitest, I noticed there's widespread use of `setResources` and `setActive` related to mobx and mobx-state-tree. However, I can't find any documentation about where those two functions come from. Are they defined by `mobx`? `mobx-state-tree`? Any insight into why lib-classifier relies on them would be really helpful, particularly when debugging failed tests. For instance, `setResources` no longer works as intended in this `with showTutorial` block when run with Vitest.

Reply:
https://github.com/zooniverse/front-end-monorepo/blob/9725ce6103078eb26fe3d09478e1ab0d356bf332/packages/lib-classifier/src/store/ResourceStore/ResourceStore.js#L76-L96

Reply:
Ah ok, thank you! I was confused about how every store model had access to that method, but I see `setActive` now too.

Reply:
I think I’ve still got a very old issue suggesting that the APIs for the classifier store models should be documented. At one point, I think we were going to add JSDoc descriptions to these actions.